Corrected italic n, added superscript + subscript

Theocrat opened the pull request:
In the original MIM file, the character combination "in" results in the symbol ∈ and not the italicized 𝑛, because the two combinations clashed. In the first change, the character combination for ∈ was altered to "𝐈𝐧" (with an uppercase 𝐈)

Secondly, unicode characters for superscript and subscript were incorporated into the MIM file, using "^2" for ² for instance. This allows the user to (1) write chemical equations:
2H₂O → 2H₂ + O₂ using "_2", for instance
and (2) write equations with indices:
xₙ² - x₀ⁿ = x₀⁽ⁿ⁺¹⁾
and also (3) compose n-ary operations and integrations:
Σᵢ aᵢ² = A
₀∫¹ 𝑒ⁿ·dn = e - 1

Terminating spaces may have been auto-removed from some lines by my text editor

Theocrat committed:
Corrected italic n, added superscript + subscript

In the original MIM file, the character combination "in" results in the symbol ∈ and not the italicized 𝑛, because the two combinations clashed. In the first change, the character combination for ∈ was altered to "𝐈𝐧" (with an uppercase 𝐈)

Secondly, unicode characters for superscript and subscript were incorporated into the MIM file, using "^2" for ² for instance. This allows the user to (1) write chemical equations:
2H₂O → 2H₂ + O₂ using "_2", for instance
and (2) write equations with indices:
xₙ² - x₀ⁿ = x₀⁽ⁿ⁺¹⁾
and also (3) compose n-ary operations and integrations:
Σᵢ aᵢ² = A
₀∫¹ 𝑒ⁿ·dn = e - 1

vvasuki commented on math-latex.mim:
```diff
@@ -349,6 +349,62 @@ you don't have to turn off this input method to input normal ASCII characters.")
     ("<>"       "⇔")
     ("barl"     "↦")
     ("mapsto"   "↦")
+;; Superscripts
```
Consider the s1 ... sn and v1 ... v9 codes. Given that, does this addition make sense? `[^1]` etc.. is oft used in markdown files to indicate footnotes - so these additions would inconvenience users.

Theocrat replied:
This comment made me notice the s0-s9 and v0-v9 sequences. I had not noticed them earlier, embarrassing as that is. I must have missed them while scanning the MIM file since I had already noted that the "s" key is used for script letters 𝒜-𝒵, so I assumed that the digits would be likewise treated. I think I ought to remove most of them.

However, there is a problem with keeping these sequences as-is, and there is a change still required: namely, "sn" maps to the scripts letter "𝓃", and not to the superscripted "ⁿ"; and further, that the subscript characters for letters i, a, o, etc. do not exist while the subscripts for numbers alone exist.

The subscripts are easier to handle - I can simply _append_ the additional subscript characters at the end of the subscript section On the other hand, ⁿ is harder to incorporate, since it currently clashes with 𝓃. I would suggest moving 𝓃 to "Sn" if that is all right?

vvasuki replied:
I agree with your proposed solution.

vvasuki commented on math-latex.mim:
```diff
@@ -131,7 +131,7 @@ you don't have to turn off this input method to input normal ASCII characters.")
     ("succeq"       "≽")
 
-    ("in"          "∈")
+    ("In"          "∈")
```
Given that ∈ is used far more than 𝑛 - might it be better to use In for the latter?

Theocrat replied:
Yes, that does make sense. As long as ∈ and 𝑛 have distinct keystroke sequences, the problem is solved.

vvasuki reviewed (changes requested):
Thanks - please see comments.

Theocrat commented:
This comment made me notice the s0-s9 and v0-v9 sequences. I had not
noticed them earlier, embarrassing as that is. I must have missed them
while scanning the MIM file since I had already noted that the "s" key is
used for script letters 𝒜-𝒵, so I assumed that the digits would be
likewise treated. I think I ought to remove most of them.

However, there is a problem with keeping these sequences as-is, and there
is a change still required: namely, "sn" maps to the scripts letter "𝓃",
and not to the superscripted "ⁿ"; and further, that the subscript
characters for letters i, a, o, etc. do not exist while the subscripts for
numbers alone exist.

The subscripts are easier to handle - I can simply *append* the additional
subscript characters at the end of the subscript section On the other hand,
ⁿ is harder to incorporate, since it currently clashes with 𝓃. I would
suggest moving 𝓃 to "Sn" if that is all right?
As for the matter of 𝑛 clashing with ∈, yes, I agree that ∈ is used far
more frequently, and should probably be assigned "in".

I shall incorporate these changes to the pull request then. I thank you
again for your time and responsiveness in this matter.

Rajarshi


On Fri, 20 Aug 2021 at 17:38, Vishvas Vasuki विश्वासः <
***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Thanks - please see comments.
> ------------------------------
>
> In math-latex.mim
> <https://github.com/indic-transliteration/m17n-db-indic/pull/4#discussion_r692892582>
> :
>
> > @@ -349,6 +349,62 @@ you don't have to turn off this input method to input normal ASCII characters.")
>
>      ("<>"       "⇔")
>
>      ("barl"     "↦")
>
>      ("mapsto"   "↦")
>
> +;; Superscripts
>
>
> Consider the s1 ... sn and v1 ... v9 codes. Given that, does this addition
> make sense? [^1] etc.. is oft used in markdown files to indicate
> footnotes - so these additions would inconvenience users.
> ------------------------------
>
> In math-latex.mim
> <https://github.com/indic-transliteration/m17n-db-indic/pull/4#discussion_r692894173>
> :
>
> > @@ -131,7 +131,7 @@ you don't have to turn off this input method to input normal ASCII characters.")
>
>      ("succ"       "≻")
>
>      ("succeq"       "≽")
>
>
>
> -    ("in"          "∈")
>
> +    ("In"          "∈")
>
>
> Given that ∈ is used far more than 𝑛 - might it be better to use In for
> the latter?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/indic-transliteration/m17n-db-indic/pull/4#pullrequestreview-734958808>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADHU57H4BUQ65QV3IEAGNJDT5ZAT7ANCNFSM5CP7MA5A>
> .
>

Theocrat committed:
Appending suggested changes to the pull request

I have removed the ^ and _ keyboard sequences which I had appended earlier, but now I have added a series with 𝐯 to incorporate the subscripted letters and Greek characters (lines 355-379). I have also taken the liberty to modify lines 347 and 348, to avoid any clash between ⇒ and 𝕝 (both formerly "dl"), and also between ⇐ and 𝕙 (both formerly "dh"). I have changed the combination for the italic 𝑛 to "In" to avoid clash with ∈, and the scripted 𝓃 to "Sn" the superscript ⁿ, and set the corresponding symbols ∈ and ⁿ to "in" and "sn" respectively. Finally, on lines 352 and 353, I have added two more symbols, which hope I have not added redundantly as before. These are the therefore and because (∴ and ∵) triple-dots. I ran searches for these unicodes inside my text editor and couldn't find them, so I considered it safe to add them. Do tell me if I have added what was already there in the file.

Theocrat commented:
I have incorporated the changes in the patch_1 branch of the fork. I am not
very familiar with the process of Pull Requests, so I am not sure if it has
gone through. Kindly tell me if the changes have not been made to the Pull
Request, and if something more is required of me.

On Fri, 20 Aug 2021 at 21:22, Rajarshi Bandopadhyay ***@***.***>
wrote:

> This comment made me notice the s0-s9 and v0-v9 sequences. I had not
> noticed them earlier, embarrassing as that is. I must have missed them
> while scanning the MIM file since I had already noted that the "s" key is
> used for script letters 𝒜-𝒵, so I assumed that the digits would be
> likewise treated. I think I ought to remove most of them.
>
> However, there is a problem with keeping these sequences as-is, and there
> is a change still required: namely, "sn" maps to the scripts letter "𝓃",
> and not to the superscripted "ⁿ"; and further, that the subscript
> characters for letters i, a, o, etc. do not exist while the subscripts for
> numbers alone exist.
>
> The subscripts are easier to handle - I can simply *append* the
> additional subscript characters at the end of the subscript section On the
> other hand, ⁿ is harder to incorporate, since it currently clashes with 𝓃.
> I would suggest moving 𝓃 to "Sn" if that is all right?
> As for the matter of 𝑛 clashing with ∈, yes, I agree that ∈ is used far
> more frequently, and should probably be assigned "in".
>
> I shall incorporate these changes to the pull request then. I thank you
> again for your time and responsiveness in this matter.
>
> Rajarshi
>
>
> On Fri, 20 Aug 2021 at 17:38, Vishvas Vasuki विश्वासः <
> ***@***.***> wrote:
>
>> ***@***.**** requested changes on this pull request.
>>
>> Thanks - please see comments.
>> ------------------------------
>>
>> In math-latex.mim
>> <https://github.com/indic-transliteration/m17n-db-indic/pull/4#discussion_r692892582>
>> :
>>
>> > @@ -349,6 +349,62 @@ you don't have to turn off this input method to input normal ASCII characters.")
>>
>>      ("<>"       "⇔")
>>
>>      ("barl"     "↦")
>>
>>      ("mapsto"   "↦")
>>
>> +;; Superscripts
>>
>>
>> Consider the s1 ... sn and v1 ... v9 codes. Given that, does this
>> addition make sense? [^1] etc.. is oft used in markdown files to
>> indicate footnotes - so these additions would inconvenience users.
>> ------------------------------
>>
>> In math-latex.mim
>> <https://github.com/indic-transliteration/m17n-db-indic/pull/4#discussion_r692894173>
>> :
>>
>> > @@ -131,7 +131,7 @@ you don't have to turn off this input method to input normal ASCII characters.")
>>
>>      ("succ"       "≻")
>>
>>      ("succeq"       "≽")
>>
>>
>>
>> -    ("in"          "∈")
>>
>> +    ("In"          "∈")
>>
>>
>> Given that ∈ is used far more than 𝑛 - might it be better to use In for
>> the latter?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/indic-transliteration/m17n-db-indic/pull/4#pullrequestreview-734958808>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ADHU57H4BUQ65QV3IEAGNJDT5ZAT7ANCNFSM5CP7MA5A>
>> .
>>
>

vvasuki commented:
Great, thanks!
